### ISSUE TITLE
During addresses import it would be handy if the store mapped the country ID itself.

### ISSUE THREAD

**Issue by grzegorzgit** [PrestaShop/PrestaShop#21736]
When you import csv with addresses list, there is a "country" column and this column accepts only IDs from Prestashop. It would be handy if the two letters code would be enough (so that store could map the id itself).

#### Describe the solution you'd like

In the store the countries are anyway stored with their two letters code, it would be better to use this instead of id. It would be much more user friendly, and it could do automatically the work that now has to be done by user, i.e.  exporting the countries with id and then map it by the letters code into the addresses list.

**Comment by hibatallahAouadni** [comment]
Hi @grzegorzgit 

Thanks for the suggestion, the Product Team will take a look.
Please stay tuned !

Ping @PrestaShop/product-team , what do you think?

**Comment by frenchfriez** [comment]
I totally agree with @grzegorzgit ! I am currently importing my client addresses on prestashop but I am facing errors...
![prestashop import addresses error](https://user-images.githubusercontent.com/71665846/98022032-600bfc00-1e05-11eb-934b-ac38a08efb0d.png)
Country->need_identification_number is empty
Address->id_country is empty

Did anyone have the same issue ?

I believe the "Address->id_country is empty" is due to the IDs from Prestashop. Any idea where I can find this country ID list ?

**Comment by hibatallahAouadni** [comment]
Hi @frenchfriez 

This ticket concerns a suggestion for improvement. But what you posted is an issue, so could you please open a new issue with the bug's description and the steps to reproduce it, to fix the problem.

Thanks!

**Comment by grzegorzgit** [comment]
@frenchfriez here is the list of countries iso codes with their ID that I exported from PrestaShop database, 
[countries list.zip](https://github.com/PrestaShop/PrestaShop/files/5486427/countries.list.zip)

And actually you can find it in the international > locations > countries tab

**Comment by boo-code** [comment]
Names already resolve on `9.2.x` - a string column has always gone through `Country::getIdByName()` - so the column taking only ids is not what the code does. What an ISO code hits is a failure rather than a missing convenience.

Measured on `9.2.x`:

```
input 'PL'      getIdByName=false   getByIso=14   -> falls through to the create-a-country branch
input 'FR'      getIdByName=false   getByIso=8    -> falls through to the create-a-country branch
input 'Poland'  getIdByName=14                    -> resolves
input 'France'  getIdByName=8                     -> resolves
```

and that fallback cannot succeed, because it never sets a required field:

```
Country as the import builds it  -> validateFields: 'Property Country->need_identification_number is empty.'
plus need_identification_number  -> validateFields: true
```

`Country` declares six required fields (`id_zone`, `iso_code`, `contains_states`, `need_identification_number`, `display_tax_label`, `name`) and the import sets four, so `add()` is never reached. No duplicate country is created either - worth checking first, since `country_iso_code` is not a unique index and one would otherwise have been inserted silently.

That is the failure in the screenshot posted in this thread in 2020: `Country->need_identification_number is empty` followed by `Address->id_country is empty`. It was deferred to a separate issue that does not appear to have been opened, and it is still reachable for any country name the shop does not already have, not only for ISO codes.

The store import carries the same two lines.
